File: backend/src/models/Message.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Message = sequelize.define('Message', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  account_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'email_accounts',
      key: 'id'
    }
  },
  message_id: {
    type: DataTypes.STRING(500)
  },
  from_address: {
    type: DataTypes.TEXT
  },
  to_addresses: {
    type: DataTypes.TEXT
  },
  cc_addresses: {
    type: DataTypes.TEXT
  },
  subject: {
    type: DataTypes.TEXT
  },
  body_text: {
    type: DataTypes.TEXT
  },
  body_html: {
    type: DataTypes.TEXT
  },
  received_date: {
    type: DataTypes.DATE
  },
  is_read: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  },
  is_starred: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  },
  folder: {
    type: DataTypes.STRING(100),
    defaultValue: 'INBOX'
  },
  raw_headers: {
    type: DataTypes.JSONB
  }
}, {
  tableName: 'messages',
  underscored: true,
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: false
});

module.exports = Message;

File: backend-dotnet/Mailler.Api/Models/Message.cs
using System.Text.Json;

namespace Mailler.Api.Models;

public sealed class Message
{
    public int Id { get; set; }

    public int AccountId { get; set; }

    public string? MessageId { get; set; }

    public string? FromAddress { get; set; }

    public string? ToAddresses { get; set; }

    public string? CcAddresses { get; set; }

    public string? Subject { get; set; }

    public string? BodyText { get; set; }

    public string? BodyHtml { get; set; }

    public DateTime? ReceivedDate { get; set; }

    public bool IsRead { get; set; }

    public bool IsStarred { get; set; }

    public string Folder { get; set; } = "INBOX";

    public JsonDocument? RawHeaders { get; set; }

    public DateTime CreatedAt { get; set; }

    public EmailAccount? Account { get; set; }
}